pkg/asset/agent/manifests/agentpullsecret.go

```diff
@@ -112,11 +115,17 @@ func (a *AgentPullSecret) finish() error {
 	}
 
 	// Normalise the JSON formatting so that we can redact the file reliably
-	normal, err := normalizeDockerConfig(a.Config.StringData[pullSecretKey])
+	pullSecretValue, ok := a.Config.StringData[pullSecretKey]
+	if !ok {
+		pullSecretValue = string(a.Config.Data[pullSecretKey])
+	}
+	normal, err := normalizeDockerConfig(pullSecretValue)
```

Comment:
Extracting the field with the effective secret data could sound as a good target for refactoring a small func, ie something like an internal version of `GetPullSecretData` but working directly on a Secret object